app/components/agenda-manager/index.hbs

```diff
@@ -24,15 +29,18 @@
       </AgendaManager::AgendaTable>
 
     {{/if}}
-
   </div>
-  <AgendaManager::Edit @visible={{this.editModalVisible}}
-                       @itemToEdit={{this.itemToEdit}}
-                       @saveTask={{agenda.saveItemTask}}
-                       @onClose={{this.closeModal}}
-                       @deleteTask={{agenda.deleteItemTask}}
-                       @onCancel={{this.closeModal}}
-                       @agendaItems={{agenda.items}}
-  />
+
+  {{#if this.editModalVisible}}
```

Comment:
I prefer not rendering components at all when they aren't needed / used. (This also means the VM has less work to do, so it should have a slight impact on rendering performance.)